src/Demo.js

```diff
@@ -0,0 +1,173 @@
+     * This event handler will dispatch an updateAnchor event to the Phaser.Game instance.
+     * @param {Object.<string, number>} e 
+     */
+    onDemoAnchorChange(e) {
+        this.anchor.direction.x = getElse([e, "data", "direction", "x"], DEFAULTS.anchor.direction.x);
+        this.anchor.direction.y = getElse([e, "data", "direction", "y"], DEFAULTS.anchor.direction.y);
+        this.anchor.position.x = getElse([e, "data", "position", "x"], DEFAULTS.anchor.position.x);
+        this.anchor.position.y = getElse([e, "data", "position", "y"], DEFAULTS.anchor.position.y);
+
+        // If the game has been initialized, then forward the updateAnchor event.
+        if (this.game !== undefined) {
+            this.game.events.emit("updateAnchor");
+            // Force a window resize event so the anchors refresh.
+            window.dispatchEvent(new Event("resize"));
+        }
+    }
+
+    /**
+     * Handler for when the SafeScaleManager detects a resize event on the window.
+     * @param {*} param0 
+     */
+    onSafeScaleResize({ scaleRatio }) {
+        if (!this.game || !this.game.canvas) {
+            return;
+        }
+
+        // Scale the size of the game canvas based off the ratio provided by the SafeScaleManager.
+        this.game.canvas.style.width = `${this.resolutions.maxWidth * scaleRatio}px`;
+        this.game.canvas.style.height = `${this.resolutions.maxHeight * scaleRatio}px`;
+    }
+
+    /**
+     * Initializes a new Phaser.Game instance.
+     */
+    initializeGame() {
+        if (this.game !== undefined) {
+            this.game.destroy(true);
+        }
+
+        // Create a new game at the max resolution.
+        this.game = new Phaser.Game({
+            width: this.resolutions.maxWidth,
+            height: this.resolutions.maxHeight,
+            scene: GameScene
+        });
+
+        // Force a window resize event.
+        window.dispatchEvent(new Event("resize"));
+    }
+
+    /**
+     * Processes the changed demo resolutions to make sure the max resolution is larger than the safe resolution.
+     */
+    validateResolutions() {
+        // Note: These min/max checks happen in the SafeScaleManager constructor.
+        //       However because we don't create a new safe scale manager, we need to manually
+        //       do these checks when the demo resolutions change.
+
+        const maxWidth = Math.max(this.resolutions.maxWidth, this.resolutions.safeWidth);
+        const maxHeight = Math.max(this.resolutions.maxHeight, this.resolutions.safeHeight);
+        const minWidth = Math.min(this.resolutions.maxWidth, this.resolutions.safeWidth);
+        const minHeight = Math.min(this.resolutions.maxHeight, this.resolutions.safeHeight);
+
+        this.resolutions.maxWidth = maxWidth;
+        this.resolutions.maxHeight = maxHeight;
+        this.resolutions.safeWidth = minWidth;
+        this.resolutions.safeHeight = minHeight;
+    }
+}
+
+function getElse(path, fallback) {
```

Comment:
I think these function calls might be a tad more intuitive if they read like this instead:
```javascript
getElse(object, 'path.to.x', defaultValue);
```

shouldn't be a gigantic refactor but I think it'll help a little bit (so people don't have to scroll down to see the function definition).

I'm also wondering if there's a more descriptive name here. `getElse` is a tad sparse in meaning for me, but I can't think of another name...